src/tools/tools/LoggerCSharp.fs

```diff
@@ -47,6 +47,18 @@ type Logger private() =
         if Logger.LoggerProvider.IsEnabled(logId, logLevel) then
             Logger.LoggerProvider.Log((logId, logLevel, messageFunc.Invoke()))
 
+    /// Log message generated by "messageFunc" using "JobId", 
+    /// "messageFunc" is evaluated only if "logLevel" is less than or equal to the log level for "logId"
+    static member inline LogF(jobID : Guid, logLevel : LogLevel,messageFunc : Func<string>) =
+        if logLevel <= Logger.DefaultLogLevel then
+            Logger.LoggerProvider.Log((null,jobID, logLevel, messageFunc.Invoke()))
```

Comment:
Once we have an overload of  "(Guid * LogLevel * String) -> unit", then there is no need to pass "null" for the first parameter.